.github/workflows/clang-tidy.yaml

```diff
@@ -32,24 +30,23 @@ jobs:
   clang-tidy:
     name: Run clang-tidy
     runs-on: ubuntu-latest
+    # be consistent with tools/toolchain/image
+    container: scylladb/scylla-toolchain:fedora-40-20240621
     steps:
       - uses: actions/checkout@v4
         with:
           submodules: true
-      - uses: ./.github/actions/setup-build
-        with:
-          install_clang_tidy: true
+      - run: |
+          sudo dnf -y install clang-tools-extra
```

Comment:
Side question: why did we use ubuntu for running clang-tidy, and why do we still use it for iwyu?

Reply:
I guess the answer is here:

> since we are have clang 18 in fedora 40

I suppose this is because we moved to fedora 40 only recently, and the older toolchain didn't have clang 18 which was apparently needed for these checks.

Reply:
because i wanted to use the latest stable release of clang in hope that we could enjoy all the latest features and fixes, and by then there were two options

- https://apt.llvm.org/
- https://copr.fedorainfracloud.org/coprs/g/fedora-llvm-team/llvm-snapshots/

i thought that the former might be more reliable as it was maintained by llvm team, and it might have better CDN.

regarding iwyu, i haven't got a chance to update it. as i thought this issue was critical, so wanted to create the fix ASAP.

Reply:
created #19655